nx/lib/nx/pytorch_backend.ex

```diff
@@ -0,0 +1,177 @@
+defmodule Nx.PytorchBackend do
+  @behaviour Nx.Backend
+
+  defstruct [:ref]
+
+  alias Nx.Pytorch.NIF
+
+  import Nx.Shared
+
+  funs =
+    Nx.Backend.behaviour_info(:callbacks) --
+      [
+        tensor: 1,
+        iota: 2,
+        eye: 1,
+        random_uniform: 3,
+        random_normal: 3,
+        reshape: 3,
+        squeeze: 3,
+        add: 3,
+        dot: 5,
+        cholesky: 2,
+        from_binary: 3,
+        to_binary: 2,
+        backend_deallocate: 1,
+        inspect: 2
+      ]
+
+  for {fun, arity} <- funs do
+    args = Macro.generate_arguments(arity, __MODULE__)
+
+    def unquote(fun)(unquote_splicing(args)) do
+      raise "#{unquote(fun)}() is not supported"
+    end
+  end
+
+  ## Creation
+
+  @impl true
+  def tensor(tensor) do
+    tensor
+  end
+
+  def torch_type({:u, 8}), do: :byte
+  def torch_type({:s, 8}), do: :char
+  def torch_type({:s, 16}), do: :short
+  def torch_type({:s, 32}), do: :int
+  def torch_type({:s, 64}), do: :long
+  def torch_type({:f, 16}), do: :half
+  def torch_type({:f, 32}), do: :float
+  def torch_type({:f, 64}), do: :double
+
+  @impl true
+  def iota(out, axis \\ nil)
+
+  def iota(%{shape: {}, type: type} = out, nil), do: iota(%{out | shape: {1}}, nil)
```

Comment:
This will unfortunately return a tensor with the wrong shape (the shape will be {1}, it should be {}).